### PR TITLE
feat: remote write

### PR DESCRIPTION
**Pull request by srodenhuis**
partially-implements: #962
Enables Prometheus to be configured for remote write.
Note: is a prerequisite for upcoming Thanos PR

### CODE REVIEW

File: tests/fixtures/env/apps/prometheus.yaml [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+            rwConfig:
+                basicAuth:
+                    enabled: true
+                    username: testaccount
+                customConfig: |-
+                    writeRelabelConfigs:
+                      - targetLabel: tenant
+                        sourceLabels:
+                        - instance
+                        replacement: otomi-aks-ont
```
Comment (j-zimnowoda) [inline tests/fixtures/env/apps/prometheus.yaml:16]: shouldn't it be a cluster.name or cluster.domainSuffix ?

File: values/prometheus-operator/prometheus-operator.gotmpl [hunk 2]
```diff
@@ -102,6 +102,26 @@ prometheus:
...
+          name: prometheus-remote-write-basic-auth
+          key: username
+        password:
+          name: prometheus-remote-write-basic-auth
+          key: password
+      {{- end }}
+      tlsConfig:
+        insecureSkipVerify: true
+      {{- if $p.remoteWrite.otomiThanos }}
+      headers:
```
Comment (j-zimnowoda) [inline values/prometheus-operator/prometheus-operator.gotmpl:121]: why the `cluster` header is needed for Thanos?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Looking good.
Could you please refer to my comments?

File: values/prometheus-operator/prometheus-operator.gotmpl (hunk 2 above)
Comment (srodenhuis) [reply values/prometheus-operator/prometheus-operator.gotmpl:121]: To label data coming from a specific cluster

File: values-schema.yaml [hunk 3]
```diff
@@ -2224,6 +2224,34 @@ properties:
           enabled:
             type: boolean
             default: false
+          remoteWrite:
+            properties:
+              enabled:
+                type: boolean
+                default: false
+              otomiThanos:
```
Comment (j-zimnowoda) [inline values-schema.yaml:2232]: Thanos is not delivered yet, so lets remove this property.

**Review by j-zimnowoda — APPROVED** [review]

**Pull request merged**